src/charm.py

```diff
@@ -176,37 +233,9 @@ def _configure_amf(self, event: EventBase) -> None:  # noqa C901
         Args:
             event (PebbleReadyEvent, DatabaseCreatedEvent, NRFAvailableEvent): Juju event
         """
-        if not self._amf_container.can_connect():
-            self.unit.status = MaintenanceStatus("Waiting for service to start")
-            event.defer()
-            return
-        if invalid_configs := self._get_invalid_configs():
-            self.unit.status = BlockedStatus(
-                f"The following configurations are not valid: {invalid_configs}"
-            )
-            return
-        for relation in ["fiveg-nrf", "database", "certificates"]:
-            if not self._relation_created(relation):
-                self.unit.status = BlockedStatus(f"Waiting for {relation} relation")
-                return
-        if not self._database_is_available():
-            self.unit.status = WaitingStatus("Waiting for the amf database to be available")
-            event.defer()
-            return
-        if not self._get_database_info():
-            self.unit.status = WaitingStatus("Waiting for AMF database info to be available")
-            event.defer()
-            return
-        if not self._nrf_requires.nrf_url:
-            self.unit.status = WaitingStatus("Waiting for NRF data to be available")
-            event.defer()
-            return
-        if not self._amf_container.exists(path=CONFIG_DIR_PATH):
-            self.unit.status = WaitingStatus("Waiting for storage to be attached")
-            event.defer()
-            return
-        if not _get_pod_ip():
-            self.unit.status = WaitingStatus("Waiting for pod IP address to be available")
+        if self._is_unit_in_non_active_status():
+            # Unit Status is in Maintanence or Blocked or Waiting status
+            self.unit.status = self._is_unit_in_non_active_status()  # type: ignore
```

Comment:
Normally you wouldn't do this again here -- you're already observing `collect_unit_status` so that will evaluate the current status after every event. Or, if you are sure you need to defer those events (I'm not sure we want defer there, though) you would evaluate the status but only use it for calling defer (not set the status):

```python
if self._is_unit_in_non_active_status():
    event.defer()
    return
```

Reply:
Done